### ISSUE TITLE
OS Browser Plugin/Extension

### ISSUE THREAD

**Issue by Arinerron** [OpenSprites/OpenSprites-next#26]
Credit to @nanalan:

How about we have a browser extension that adds a button to the OpenSprites site next to the download one to "Add to Backpack". Then it uses the session id from Scratch so we don't have to ask for the user and password of the Scratch account to add the resource to the user's backpack.

I can do the chrome extension, but not Firefox-- I have no experience in Firefox's extension development.

**Comment by bates64** [comment]
> I can do the chrome extension, but not Firefox-- I have no experience in Firefox's extension development.

Same here

**Comment by bates64** [comment]
Also we could have a "share from backpack" button that allows us to read their backpack as files - this is blocked by #28 though

**Comment by MegaApuTurkUltra** [comment]
So the way this would probably work is we'd load Scratch in an iframe on OpenSprites and `postMessage` to it, the extension would hook into Scratch pages, read posted messages, and would be able to ajax with Scratch's session id

**Comment by MegaApuTurkUltra** [comment]
Features we might want
- Sync OS backpack with Scratch backpack
- Upload projects from OpenSprites

**Comment by towerofnix** [comment]
Is this really necessary for initial release? While I totally support this feature, directly uploading from your downloads folder isn't _that_ hard (and also a useful skill for people to have, should we teach it to them!). :P

**Comment by andrewjcole** [comment]
Nope, I don't think so. Feel free to put it back to Initial Release if you think I'm wrong.
